### DISCUSSION TITLE
Custom interval for reconnecting after connection loss

### DISCUSSION THREAD

**Discussion by dariusptrs** [goToMain/libosdp#198] [Q&A] [answered]
When the connection goes away, I get this message:

`OSDP: CP: PD-1: 2024-10-23T13:57:45Z osdp_cp.c:1211 [ERROR] Going offline for 300 seconds; Was in 'ID-Request' state.`

It would be great if the `300 seconds` value could be configurable by the user.

Additionally, having the ability to detect this event (e.g., triggering an event in the code) would be helpful for error handling.

I'm using C++. If there’s already a way to achieve this, I'd appreciate some guidance!

**Comment by sidcha** [comment, marked as answer]
> It would be great if the 300 seconds value could be configurable by the user.

You can customize this timeout by editing the src/config.h.in file. For this case, it would be [osdp_config.h.in:31](https://github.com/goToMain/libosdp/blob/master/src/osdp_config.h.in#L31)

> Additionally, having the ability to detect this event (e.g., triggering an event in the code) would be helpful for error handling.

See notification [events](https://github.com/goToMain/libosdp/blob/master/include/osdp.h#L853); also don't forget to [enable](https://github.com/goToMain/libosdp/blob/master/include/osdp.h#L63) it for each PD that you want to receive those notifications.
